File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.java
// Last updated: 6/8/2025, 11:54:38 AM
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] arr = new int[nums.length];
        for(int i = 0 ; i <nums.length ; i++){
            arr[i] = 1;
        }
        int left = 1;
        for(int i =0 ; i <nums.length ; i++){
            arr[i] *= left;
            left *= nums[i];
        }

        int right = 1 ;
        for(int i =nums.length - 1 ; i  >=0 ; i--){
            arr[i] *= right;
            right *= nums[i];
        }
        return arr;
    }
}

File: 238-ProductOfArrayExceptSelf/238-ProductOfArrayExceptSelf.py
# Last updated: 6/8/2025, 11:54:36 AM
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [1] * n
        left = 1
        
        for l in range(n):
            ans[l]*= left
            left *=nums[l]
        right = 1
        for l in range(n-1, -1,-1):
            ans[l]*= right
            right *=nums[l]

        return ans
